stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/MessageList.kt

```diff
@@ -223,26 +233,56 @@ public fun Messages(
     }
 }
 
+/**
+ * Shows an option when the user scrolls away from the bottom of the list. If there are any new messages it also gives
+ * the user information on how many messages they haven't read.
+ *
+ * @param unreadCount - The count of unread messages.
+ * @param onClick - The handler that's triggered when the user taps on the action.
+ * */
 @OptIn(ExperimentalMaterialApi::class)
 @Composable
 private fun BoxScope.MessagesScrollingOption(
-    text: String,
+    unreadCount: Int,
     onClick: () -> Unit,
 ) {
-    Surface(
+    Box(
         modifier = Modifier
-            .align(Alignment.BottomCenter)
-            .padding(16.dp),
-        shape = RoundedCornerShape(16.dp),
-        elevation = 4.dp,
-        onClick = onClick,
-        color = ChatTheme.colors.inputBackground,
+            .align(BottomEnd)
+            .padding(16.dp)
+            .wrapContentSize()
     ) {
-        Text(
-            modifier = Modifier.padding(8.dp),
-            text = text,
-            style = ChatTheme.typography.body,
-            color = ChatTheme.colors.textHighEmphasis,
-        )
+        Surface(
+            modifier = Modifier
+                .padding(top = 12.dp)
+                .size(48.dp),
+            shape = CircleShape,
+            elevation = 4.dp,
+            indication = rememberRipple(),
+            onClick = onClick
+        ) {
+            Icon(
+                modifier = Modifier.padding(16.dp),
+                painter = painterResource(R.drawable.stream_compose_ic__arrow_down),
+                contentDescription = null,
+                tint = ChatTheme.colors.primaryAccent
+            )
+        }
+
+        if (unreadCount != 0) {
+            Surface(
+                modifier = Modifier
+                    .align(TopCenter),
+                shape = RoundedCornerShape(16.dp),
+                color = ChatTheme.colors.primaryAccent
+            ) {
+                Text(
```

Comment:
[Query] Any reason to wrap it inside a Surface or to get BG color as per L#277

Reply:
Used a surface to make it a bit easier to allow for padding and a shape that wraps the text. We could add lots of these properties to the `Text` but it felt like it'd be too hard to follow the order of the modifiers.

LMK if this makes sense!